src/test/store/profile-view.test.js

```diff
@@ -116,83 +123,613 @@ describe('call node paths on implementation filter change', function() {
+    });
+  });
+
+  describe('changeSelectedThread', function() {
+    const { profile } = getProfileFromTextSamples('A', 'B');
     const { dispatch, getState } = storeWithProfile(profile);
-    const callNodeInfo = selectedThreadSelectors.getCallNodeInfo(getState());
 
-    // Before expand all action is dispatched, nothing is expanded
-    expect(
-      selectedThreadSelectors.getExpandedCallNodePaths(getState())
-    ).toEqual([]);
-
-    dispatch(
-      ProfileView.expandAllCallNodeDescendants(
-        threadIndex,
-        0, // A
-        callNodeInfo
-      )
-    );
+    it('can set and change the selected thread', function() {
+      expect(UrlStateSelectors.getSelectedThreadIndex(getState())).toEqual(0);
+      dispatch(ProfileView.changeSelectedThread(1));
+      expect(UrlStateSelectors.getSelectedThreadIndex(getState())).toEqual(1);
+    });
+  });
 
-    expect(
-      selectedThreadSelectors.getExpandedCallNodePaths(getState()).sort()
-    ).toEqual([
-      // Paths
-      [A],
-      [A, B],
-      [A, B, C],
-      [A, B, D],
-      [A, E],
-    ]);
+  describe('focusCallTree', function() {
+    const { profile } = getProfileFromTextSamples('A');
+    const { dispatch, getState } = storeWithProfile(profile);
+
+    it('updates the focus call tree generation', function() {
+      expect(
+        ProfileViewSelectors.getFocusCallTreeGeneration(getState())
+      ).toEqual(0);
+      dispatch(ProfileView.focusCallTree());
+      expect(
+        ProfileViewSelectors.getFocusCallTreeGeneration(getState())
+      ).toEqual(1);
+    });
   });
 
-  it('expands subtrees', function() {
+  describe('changeRightClickedThread', function() {
+    const { profile } = getProfileFromTextSamples('A', 'B');
     const { dispatch, getState } = storeWithProfile(profile);
 
-    // First expand A by selecting B
-    dispatch(ProfileView.changeSelectedCallNode(threadIndex, [A, B]));
+    it('changes the right clicked thread index', function() {
+      expect(
+        ProfileViewSelectors.getRightClickedThreadIndex(getState())
+      ).toEqual(0);
+      dispatch(ProfileView.changeRightClickedThread(1));
+      expect(
+        ProfileViewSelectors.getRightClickedThreadIndex(getState())
+      ).toEqual(1);
+    });
+  });
 
-    const callNodeInfo = selectedThreadSelectors.getCallNodeInfo(getState());
+  describe('changeThreadOrder', function() {
+    const { profile } = getProfileFromTextSamples('A', 'B', 'C');
+    const { dispatch, getState } = storeWithProfile(profile);
 
-    // Before expand all action is dispatched, only A is expanded
-    expect(
-      selectedThreadSelectors.getExpandedCallNodePaths(getState())
-    ).toEqual([
-      // Paths
-      [A],
-    ]);
+    it('changes the thread order', function() {
+      expect(UrlStateSelectors.getThreadOrder(getState())).toEqual([0, 1, 2]);
+      withAnalyticsMock(() => {
+        dispatch(ProfileView.changeThreadOrder([2, 1, 0]));
+        expect(self.ga.mock.calls).toEqual([
```

Comment:
you can use:
```javascript
expect(self.ga).toBeCalledWith(
  'send',
  {
    eventAction: 'change thread order',
    eventCategory: 'profile',
    hitType: 'event',
  }
);
```
This gives a better error message in case of failures.

note: with the new jest, you can also do `expect(self.ga).toMatchSnapshot()`. This would work for me too, so you can choose between both.